RFS/src/main/java/com/rfs/common/RestClient.java

```diff
@@ -28,108 +25,61 @@ public Response(int responseCode, String responseBody, String responseMessage) {
     }
 
     public final ConnectionDetails connectionDetails;
+    private final HttpClient client;
 
     public RestClient(ConnectionDetails connectionDetails) {
         this.connectionDetails = connectionDetails;
-    }
-
-    public Response get(String path, boolean quietLogging) throws Exception {
-        String urlString = connectionDetails.url + "/" + path;
-        
-        URL url = new URL(urlString);
-        HttpURLConnection conn = (HttpURLConnection) url.openConnection();
-
-        // Set the request method
-        conn.setRequestMethod("GET");
-        
-        // Set the necessary headers
-        setAuthHeader(conn);
-
-        // Enable input and output streams
-        conn.setDoOutput(true);
-
-        // Report the results
-        int responseCode = conn.getResponseCode();
-
-        String responseBody;
-        if (responseCode >= HttpURLConnection.HTTP_BAD_REQUEST) {
-            // Read error stream if present
-            try (BufferedReader br = new BufferedReader(new InputStreamReader(conn.getErrorStream(), StandardCharsets.UTF_8))) {
-                responseBody = br.lines().collect(Collectors.joining(System.lineSeparator()));
-            }
-        } else {
-            // Read input stream for successful requests
-            try (BufferedReader br = new BufferedReader(new InputStreamReader(conn.getInputStream(), StandardCharsets.UTF_8))) {
-                responseBody = br.lines().collect(Collectors.joining(System.lineSeparator()));
-            }
-        }
-
-        if (quietLogging || (responseCode == HttpURLConnection.HTTP_CREATED || responseCode == HttpURLConnection.HTTP_OK)) {
-            logger.debug("Response Code: " + responseCode + ", Response Message: " + conn.getResponseMessage() + ", Response Body: " + responseBody);
-        } else {
-            logger.error("Response Code: " + responseCode + ", Response Message: " + conn.getResponseMessage() + ", Response Body: " + responseBody);
-        }
-
-        conn.disconnect();
 
-        return new Response(responseCode, responseBody, conn.getResponseMessage());
+        this.client = HttpClient.create()
+            .baseUrl(connectionDetails.url)
+            .headers(h -> {
+                h.add("Content-Type", "application/json");
+                if (connectionDetails.authType == ConnectionDetails.AuthType.BASIC) {
```

Comment:
Design Nitpick: We are going to need a better way to handle authentication in the future such as sigv4 and other methods.  IMO this is the case for using some kind of preconstructed client for interacting with the cluster.

Reply:
It would be good for users to try to keep as much unity in our command line options as possible, but I haven't thought of a good way to do that.  I'm wary to try to mix and match different params into one object for a single Beust class.

Reply:
Probably; I think @gregschohn has some tasks created to look into auth.  Not required for the current iteration so I didn't invest the time in it.